### PR TITLE
Use Action Cache for asset storage

### LINKED ISSUE #17: Update README example config - remove CircularBlobAccess
The example config quoted in the README uses CircularBlobAccess (the keyword 'circular').

CircularBlobAccess has been [removed from bb-storage](https://github.com/buildbarn/bb-storage/commit/e6de4f65c1e35ca625b7e1f2dffe9c14833f333e). And bb-remote-access has now been updated to include those changes. The README ought to be updated to use LocalBlobAccess.

### PR DESCRIPTION
**Pull request by sdclarke**
This PR replaces #13 as the reworking done is so extensive that it doesn't make sense to overwrite the other PR.

The interface to the action cache has been moved into the storage package, the asset store can now be set up as either a local blob access or using the action cache.

The configuration for a local blob access will now look like this:

```
{
  fetcher: {
    caching: {
      fetcher: {
        'error': {
          code: 5,
          message: 'Asset Not Found',
        },
      },
    },
  },
  assetCache: {
    localBlobAccess: {
      assetStore: {
        'local': {
          keyLocationMapOnBlockDevice: {
            file: {
              path: '/storage/key_location_map',
              sizeBytes: 1024 * 1024,
            },
          },
          keyLocationMapMaximumGetAttempts: 8,
          keyLocationMapMaximumPutAttempts: 32,
          oldBlocks: 8,
          currentBlocks: 24,
          newBlocks: 3,
          blocksOnBlockDevice: {
            source: {
              file: {
                path: '/storage/blocks',
                sizeBytes: 100 * 1024 * 1024,
              },
            },
            spareBlocks: 3,
          },
          persistent: {
            stateDirectoryPath: '/storage/persistent_state',
            minimumEpochInterval: '5m',
          },
        },
      },
      contentAddressableStorage:
        common.blobstore.contentAddressableStorage,
    },
  global: common.global,
  grpcServers: [{
    listenAddresses: [':4401'],
    authenticationPolicy: { allow: {} },
  }],
  allowUpdatesForInstances: ['remote', ''],
  maximumMessageSizeBytes: 16 * 1024 * 1024,
}
```

Whilst an action cache based one would be the same apart from the `assetCache` section, which would look as below:
```
  assetCache: {
    actionCache: {
      blobstore: common.blobstore,
    },
  },
```

In the current implementation, the CAS is specified once in the assetCache section, and this definition is shared with the `httpFetcher`. This may need to be altered to allow for the CAS to be specified without having to have an assetStore, perhaps by wrapping the `AssetCacheConfiguration` specified the the `ApplicationConfiguration` message in a `oneof`, to get: 
```
oneof <name> {
  AssetCacheConfiguration asset_cache = 9;

  buildbarn.configuration.blobstore.BlobAccessConfiguration
      content_addressable_storage = 10;
} 
```
I would definitely appreciate feedback on this idea before this is merged.

This PR depends on #15 and includes a slighlty modified #7.

This closes #17

### CODE REVIEW

**Comment by sdclarke** [comment]
I've just noticed [these lines](https://github.com/buildbarn/bb-remote-asset/pull/16/files#diff-10b384de80cdc65811e4dbba9c50cef046224fd7b33c8c3aa7e08a004afdc95fR34-R36) which will need removing before merging as they are now unused.

**Review by tomcoldrick-ct — COMMENTED** [review]
Nice one @sdclarke, this looks great! Apologies for how long it's taken me to review this.

I've left a couple of nits on naming, but implementation wise this looks good. If I may request one other thing, could you update the README to use the new configuration layout too?

File: pkg/storage/action_cache_asset_store.go [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+		}
+	}
+	return &asset.Asset{
+		Digest:      digest,
+		ExpireAt:    getDefaultTimestamp(),
+		LastUpdated: a.ExecutionMetadata.QueuedTimestamp,
+	}, nil
+}
+
+func (rs *actionCacheAssetStore) Get(ctx context.Context, ref *asset.AssetReference, instance digest.InstanceName) (*asset.Asset, error) {
```
Comment (tomcoldrick-ct) [inline pkg/storage/action_cache_asset_store.go:82]: It's a little hard to follow what's going on here (and more so in `Put`) due to all the type-changing and error-handling boiler plate. It might improve things to add a couple comments on how it works? In particular the construction of the `AssetReference` file node could do with a quick explanation.

**Review by tomcoldrick-ct — APPROVED** [review]
Thanks @sdclarke, this looks great!
